### PR TITLE
Allow `enabled` config, for use by charts depending on this chart conditionally

### LINKED ISSUE #3160: Update values schema to allow key `enabled`
### Proposed change

Update values schema to allow key `enabled` so the z2jh helm chart can be included as a dependency in other helm charts and be enabled using the preffered syntax

Usually, when including a chart as a helm dependency, the chart can be turned on and off by setting the charts `enabled` key to `true` or `false`.

This is the preffered method according to the [helm docs](https://helm.sh/docs/chart_best_practices/dependencies/#conditions-and-tags)

For example, if jupyterhub is a helm dependency, then it would look like this in the parent chart's `values.yaml` file:
```yaml
jupyterhub:
    enabled: true
    hub:
        ...
```

As of now, if you try to include the `enabled` key in a values.yaml, the jupyterhub values scheme will throw a validation error:
```
failed to upgrade release '<name>' from chart '<chart>': values don't meet the specifications of the schema(s) in the following chart(s):
jupyterhub:
- (root): Additional property enabled is not allowed
```

The values schema should be updated to allow for the enabled key so z2jh can be used as a dependency

### Alternative options

Unsure

### Who would use this feature?

Users who want to incorporate jupyterhub as a dependency in their own helm charts

### (Optional): Suggest a solution

Solution in PR #3162

### PR DESCRIPTION
**Pull request by monoakg**
Linked issue #3160 - Update values schema to allow key `enabled`

Closes #3160

### CODE REVIEW

File: jupyterhub/values.schema.yaml [hunk 1]
```diff
@@ -27,6 +27,11 @@ required:
   - rbac
   - global
 properties:
+  enabled:
+    type: [boolean, "null"]
+    description: |
+      enabled allows you to enable or disable jupyterhub using the preffered method
```
Comment (manics) [inline jupyterhub/values.schema.yaml:33]:
```suggestion
      enabled allows you to enable or disable jupyterhub using the preferred method
```

File: jupyterhub/values.yaml [hunk 2]
```diff
@@ -3,6 +3,9 @@
 fullnameOverride: ""
 nameOverride:
 
+# enable or disable jupyterhub if it is included as a dependency in a parent helm chart
+enabled:
```
Comment (manics) [inline jupyterhub/values.yaml:7]: What do other charts typically do? Do they leave this null, or default to `true`?

Comment (consideRatio) [reply jupyterhub/values.yaml:7]: I suggest its made null as default here, true/false has no meaning by this chart anyhow so making it null seems less confusing, and both false/true and can be defaults for a chart depending on this - such chart should define enabled to false/true explicitly.

Comment (consideRatio) [reply jupyterhub/values.yaml:7]: Most charts dont need an entry for this, but we have a strict schema.

File: jupyterhub/values.schema.yaml [hunk 3]
```diff
@@ -27,6 +27,11 @@ required:
   - rbac
   - global
 properties:
+  enabled:
+    type: [boolean, "null"]
+    description: |
+      enabled allows you to enable or disable jupyterhub using the preferred method
+      if it is included as a dependency in a parent helm chart
```
Comment (consideRatio) [inline jupyterhub/values.schema.yaml:34]:
```suggestion
      `enabled` is ignored by the jupyterhub chart itself, but a chart depending
      on the jupyterhub chart conditionally can make use this config option as
      the condition.
```

File: jupyterhub/values.yaml [hunk 4]
```diff
@@ -3,6 +3,9 @@
 fullnameOverride: ""
 nameOverride:
 
+# enable or disable jupyterhub if it is included as a dependency in a parent helm chart
```
Comment (consideRatio) [inline jupyterhub/values.yaml:6]:
```suggestion
# enabled is ignored by the jupyterhub chart itself, but a chart depending on
# the jupyterhub chart conditionally can make use this config option as the
# condition.
```

**Review by consideRatio — APPROVED** [review]
Thanks @monoakg!!

From experience and examples [from helm's documentation](https://helm.sh/docs/topics/charts/#tags-and-condition-fields-in-dependencies), its clear that `enabled` indeed is the common naming choice for this.

I've updated the language to make it really clear that this is a feature implemented in a chart conditionally depending on this chart

**Pull request merged**